src/hotspot/share/jfr/jni/jfrJavaSupport.hpp

```diff
@@ -100,7 +100,7 @@ class JfrJavaSupport : public AllStatic {
   static bool set_handler(jobject clazz, jobject handler, TRAPS);
 
   // critical
-  static void abort(jstring errorMsg, TRAPS);
+  static void abort(jstring errorMsg, TRAPS, bool dump_core=true);
```

Comment:
Not sure if this is necessary. The existing core dump logic already handles the case where a core file cannot be generated due to disk full.

Reply:
Thank you for your review.  

Whether or not hotspot generate a core file is determined by the argument of vm_abort(bool dump_core).  If the argument is "true", vm_abort(bool dump_core) calls os::abort(bool dump_core) to generate a core file. 
See the following code:
https://github.com/openjdk/jdk/blob/3c160ab5bec0c2364ec3f43c5a5789098d4699e5/src/hotspot/share/runtime/java.cpp#L625

I think JfrJavaSupport::abort() should pass "false" as an argument to vm_abort(bool dump_core).

Reply:
Ok. My point was that the os won't be able to create a core file if there is no available space.

But this is indeed more succinct, if we don't want to create a core categorically from this location.

Reply:
Just an observation but the filesystem that is full, and the filesystem where a core would be written, need not be the same file system. That said, a core dump in this case seems unwarranted.